### DISCUSSION TITLE
x-sort:handle does not work if defined in a template

### DISCUSSION THREAD

**Discussion by iamsergii** [alpinejs/alpine#4482] [7. Bugs (archived)] [answered]
in following example x-sort:handle won't be picked up by x-sort directive and the whole item would be draggable.
```html
<div x-sort>
                <template x-for="item in items" :key="item.id">
                    <div  x-sort:item="item.id">
                        <span x-text="item.text"></span>
                        <span x-sort:handle>
                            -
                        </span>
                    </div>
                </template>
                
</div>
```

related to the check in [packages/sort/src/index.js#L28](https://github.com/alpinejs/alpine/blob/36110b619eec982eb10efd41250757716f4847a1/packages/sort/src/index.js#L28), because no actual item exist at the time of initialisation.

can be patched with adding hidden tag with [x-sort:handle] outside of template.

**Comment by SimoTod** [comment, marked as answer]
You can also pass the configuration on the main tag instead of creating a fake hidden element: https://alpinejs.dev/plugins/sort#custom-configuration

```html
<div x-sort x-sort:config="{ handle: '[x-sort:handle]' }">
```

**Comment by SimoTod** [comment]
I've created https://github.com/alpinejs/alpine/pull/4483 to support it 👍
